sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuite.scala

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.sql.{QueryTest, Row}
+import org.apache.spark.sql.catalyst.analysis.{AnalysisTest, UnresolvedNamespace}
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser.parsePlan
+import org.apache.spark.sql.catalyst.plans.logical.{ShowTables, ShowTableStatement}
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{BooleanType, StringType, StructType}
+
+trait ShowTablesSuite extends QueryTest with SharedSparkSession with AnalysisTest {
+  protected def catalog: String
```

Comment:
A simpler way to test v1 vs v2 is: In v2 test, register a v2 catalog and set it as the default. Then in the test cases we just need to write `$namespace.$table`

Reply:
yeh, I thought about this but wanted to avoid dependency from a global variable. Also it will make harder to write tests that:
- check `USE ...`
- check both dsv1 and dsv2 in the same test

And with explicit catalog, it should be easier to mix the common trait to other test suite like JDBC or Hive catalogs.

Reply:
OK makes sense.